flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/table/test/SortITCase.scala

```diff
@@ -0,0 +1,121 @@
+import org.apache.flink.test.util.MultipleProgramsTestBase.TestExecutionMode
+import org.apache.flink.test.util.TestBaseUtils
+import org.junit._
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[Parameterized])
+class SortITCase(
+    mode: TestExecutionMode,
+    configMode: TableConfigMode)
+  extends TableProgramsTestBase(mode, configMode) {
+
+  @Test
+  def testOrderByDesc(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+    
+    val t = env.fromElements((1, "First"), (3, "Third"), (2, "Second")).toTable(tEnv)
+      .orderBy('_1.desc)
+
+    val expected = "3,Third\n2,Second\n1,First"
+    val results = t.toDataSet[Row].setParallelism(1).collect()
+    compareOrderedResultAsText(expected, results)
+  }
+
+  @Test
+  def testOrderByAsc(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+    
+    val t = env.fromElements((1, "First"), (3, "Third"), (2, "Second")).toTable(tEnv)
+      .orderBy('_1.asc)
+
+    val expected = "1,First\n2,Second\n3,Third"
+    val results = t.toDataSet[Row].setParallelism(1).collect()
+    compareOrderedResultAsText(expected, results)
+  }
+
+  @Test
+  def testOrderByMultipleFieldsDifferentDirections(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+    
+    env.setParallelism(2)
+    val t = env.fromElements((1, 3, "Third"), (1, 2, "Fourth"), (1, 4, "Second"),
+      (2, 1, "Sixth"), (1, 5, "First"), (1, 1, "Fifth"))
+      .toTable(tEnv).orderBy('_1.asc, '_2.desc)
+
+    val expected = "1,5,First\n1,4,Second\n1,3,Third\n1,2,Fourth\n1,1,Fifth\n2,1,Sixth"
+    val results = t.toDataSet[Row].mapPartition(rows => Seq(rows.toSeq)).collect()
+
+    implicit def rowOrdering = Ordering.by((x : Row) => (x.productElement(0).asInstanceOf[Int],
+      - x.productElement(1).asInstanceOf[Int]))
+
+    val result = results.sortBy(p => p.min).reduceLeft(_ ++ _)
+
+    compareOrderedResultAsText(expected, result)
+  }
+
+  @Test
+  def testOrderByMultipleFieldsWithSql(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+
+    val sqlQuery = "SELECT * FROM MyTable ORDER BY _1 DESC, _2 DESC"
+
+    val t = env.fromElements((1, 1, "First"), (2, 3, "Fourth"), (1, 2, "Second"),
+      (2, 1, "Third")).toTable(tEnv)
+    tEnv.registerDataSet("MyTable", t)
+
+    val queryResult = tEnv.sql(sqlQuery)
+
+    val expected = "2,3,Fourth\n2,1,Third\n1,2,Second\n1,1,First"
+    val results = queryResult.toDataSet[Row].setParallelism(1).collect()
+    compareOrderedResultAsText(expected, results)
+  }
+
+  private def compareOrderedResultAsText[T](expected: String, results: Seq[T]) = {
```

Comment:
If we always use `toDataSet[Row]` and respect the expected type in `DataSetSort`, we should not need this method because the result should always be `DataSet[Row]`